api/lcp/encrypt.py

```diff
@@ -0,0 +1,499 @@
+                not result.content_encryption_key:
+            raise LCPEncryptionException('Encryption failed')
+
+        return result
+
+    def _run_lcpencrypt_locally(self, file_path, identifier, configuration):
+        """Runs lcpencrypt using a local binary
+
+        :param file_path: File path to the book to be encrypted
+        :type file_path: string
+
+        :param identifier: Book's identifier
+        :type identifier: string
+
+        :param configuration: LCPEncryptionConfiguration instance
+        :type configuration: instance
+
+        :return: Encryption result
+        :rtype: LCPEncryptionResult
+        """
+        self._logger.info(
+            'Started running a local lcpencrypt binary. File path: {0}. Identifier: {1}'.format(
+                file_path, identifier
+            )
+        )
+
+        parameters = LCPEncryptor.Parameters(file_path, identifier, configuration)
+
+        try:
+            if parameters.output_file_path:
+                self._logger.info('Creating a directory tree for {0}'.format(parameters.output_file_path))
+
+                output_directory = os.path.dirname(parameters.output_file_path)
+
+                if not os.path.exists(output_directory):
+                    os.makedirs(output_directory)
+
+                self._logger.info('Directory tree {0} has been successfully created'.format(output_directory))
+
+            self._logger.info('Running lcpencrypt using the following parameters: {0}'.format(parameters.to_array()))
+
+            output = subprocess.check_output(parameters.to_array())
+            result = self._parse_output(output)
+        except Exception as exception:
+            self._logger.exception('An unhandled exception occurred during running a local lcpencrypt binary')
+
+            raise LCPEncryptionException(exception.message, inner_exception=exception)
+
+        self._logger.info(
+            'Finished running a local lcpencrypt binary. File path: {0}. Identifier: {1}. Result: {2}'.format(
+                file_path, identifier, result
+            )
+        )
+
+        return result
+
+    def encrypt(self, db, file_path, identifier):
+        """Encrypts a book
+
+        :param db: Database session
+        :type db: sqlalchemy.orm.session.Session
+
+        :param file_path: File path to the book to be encrypted
+        :type file_path: string
+
+        :param identifier: Book's identifier
+        :type identifier: string
+
+        :return: Encryption result
+        :rtype: LCPEncryptionResult
+        """
+        with self._configuration_factory.create(
+                self._configuration_storage, db, LCPEncryptionConfiguration) as configuration:
+            if self._lcpencrypt_exists_locally(configuration):
+                result = self._run_lcpencrypt_locally(file_path, identifier, configuration)
+
+                return result
+            else:
+                raise NotImplementedError()
```

Comment:
`NotImplementedError` is for cases where an abstract or unfinished method is called. Here the code is implemented but the configuration isn't complete, so a `ValueError` or `core.config.CannotLoadConfiguration` would be a more appropriate exception.

Reply:
In the beginning I was planning to add support for running `lcpencrypt` in a Docker container, this `else` branch should cover this case

Reply:
OK, just make sure there's a ticket for finishing this work.